internal/r8/deobfuscator_test.go

```diff
@@ -18,41 +18,279 @@
+		originalFunction:  "c",
+		originalClassname: "n8.h",
+	})
+	verifyFrame(t, (*stacktrace)[4], FrameValidation{
+		updated: true,
+		function: `okhttp3.internal.connection.ExchangeFinder.findConnection
+findHealthyConnection`,
+		classname:         "okhttp3.internal.connection.ExchangeFinder",
+		originalFunction:  "a",
+		originalClassname: "n8.d",
+	})
+	verifyFrame(t, (*stacktrace)[5], FrameValidation{
+		updated: true,
+		function: `okhttp3.internal.connection.ExchangeFinder.find
+okhttp3.internal.connection.RealCall.initExchange$okhttp
+intercept`,
+		classname:         "okhttp3.internal.connection.ConnectInterceptor",
+		originalFunction:  "intercept",
+		originalClassname: "n8.a",
+	})
+	verifyFrame(t, (*stacktrace)[6], FrameValidation{
+		updated:           true,
+		function:          "proceed",
+		classname:         "okhttp3.internal.http.RealInterceptorChain",
+		originalFunction:  "b",
+		originalClassname: "o8.f",
+	})
+	verifyFrame(t, (*stacktrace)[7], FrameValidation{
+		updated:           true,
+		function:          "intercept",
+		classname:         "okhttp3.internal.cache.CacheInterceptor",
+		originalFunction:  "intercept",
+		originalClassname: "l8.a",
+	})
+	verifyFrame(t, (*stacktrace)[8], FrameValidation{
+		updated:           true,
+		function:          "proceed",
+		classname:         "okhttp3.internal.http.RealInterceptorChain",
+		originalFunction:  "b",
+		originalClassname: "o8.f",
+	})
+	verifyFrame(t, (*stacktrace)[9], FrameValidation{
+		updated:           true,
+		function:          "intercept",
+		classname:         "okhttp3.internal.http.BridgeInterceptor",
+		originalFunction:  "intercept",
+		originalClassname: "o8.a",
+	})
+	verifyFrame(t, (*stacktrace)[10], FrameValidation{
+		updated:           true,
+		function:          "proceed",
+		classname:         "okhttp3.internal.http.RealInterceptorChain",
+		originalFunction:  "b",
+		originalClassname: "o8.f",
+	})
+	verifyFrame(t, (*stacktrace)[11], FrameValidation{
+		updated:           true,
+		function:          "intercept",
+		classname:         "okhttp3.internal.http.RetryAndFollowUpInterceptor",
+		originalFunction:  "intercept",
+		originalClassname: "o8.h",
+	})
+	verifyFrame(t, (*stacktrace)[12], FrameValidation{
+		updated:           true,
+		function:          "proceed",
+		classname:         "okhttp3.internal.http.RealInterceptorChain",
+		originalFunction:  "b",
+		originalClassname: "o8.f",
+	})
+	verifyFrame(t, (*stacktrace)[13], FrameValidation{
+		updated:           true,
+		function:          "intercept",
+		classname:         "co.elastic.apm.opbeans.app.di.ApplicationModule$provideOkHttpClient$$inlined$-addInterceptor$1",
+		originalFunction:  "intercept",
+		originalClassname: "b3.a",
+	})
+}
+
+func verifyFrame(t *testing.T, frame *model.StacktraceFrame, validation FrameValidation) {
+	// Line numbers shouldn't change after the mapping.
+	validation.lineno = (*frame).Lineno
```

Comment:
Based on @kruskall suggestion of verifying line numbers too, I figured this would be the most straightforward way of checking that the line number doesn't change after the deobfuscation process. In line `285`, we check that the "raw" line number hasn't changed, since the same one is present in the `StacktraceFrame.Lineno` field. If it were changed, then this verification would fail, as well as the one done in line `282`, because, for every field that gets changed, its original value must be moved over to the `StacktraceFrame.Original.[FieldName]` value.

This is the most straightforward solution I found based on the fact that `Lineno` must never change, however, I'm not sure if it's ideal since I'm not too familiar with the language, so I'm open to suggestions.

Reply:
I'd argue that the usefulness of line 285 and 293 is limited as this is an equiv of 
```
x = 1
assert x == 1
```

A slightly better way may be passing in the line number when we initialize the FrameValidation struct, e.g. line 190, and remove line 285.

@kruskall thoughts?

Reply:
I see. I didn't read the specification, and at first, I also thought that the obfuscation process would (similarly to JS source map) change the line numbers, though after taking a deeper look at R8's inputs and outputs I noticed that the line numbers never change, that's why I'm assuming R8 doesn't change them, so our deobfuscator doesn't do anything about changing any line numbers at the moment (at least not until we see an example where it could change them).

Reply:
@LikeTheSalad in that case do you mind passing in the line number when we initialize the FrameValidation struct, e.g. line 190, and remove line 285? I think that's slightly preferable over assigning it and asserting it right away